trestle/transforms/utils/tanium_helper.py

```diff
@@ -103,206 +83,361 @@ def __init__(self, tanium_row: t_tanium_row, comply, default_timestamp: t_timest
+        if self._cpus_max > os.cpu_count():
+            self._cpus_max = os.cpu_count()
+        if self._cpus_max < 1:
+            self._cpus_max = 1
+        # cpus min: default, max, min
+        if cpus_min is None:
+            self._cpus_min = 1
+        else:
+            self._cpus_min = int(cpus_min)
+        if self._cpus_min > self._cpus_max:
+            self._cpus_min = self._cpus_max
+        if self._cpus_min < 1:
+            self._cpus_min = 1
+
+    def _is_duplicate_component(self, rule_use: RuleUse) -> bool:
+        """Check for duplicate component."""
+        retval = False
+        component_type = rule_use.component_type
+        component_title = rule_use.component
+        component_description = rule_use.component
+        for component in self._component_map.values():
+            if component.type == component_type:
+                if component.title == component_title:
+                    if component.description == component_description:
+                        retval = True
+                        break
+        return retval
+
+    def _derive_components(self) -> Dict[str, ValuesView[InventoryItem]]:
+        """Derive components from RuleUse list."""
+        self._component_map = {}
+        for rule_use in self._rule_use_list:
+            if self._is_duplicate_component(rule_use):
+                continue
+            component_type = rule_use.component_type
+            component_title = rule_use.component
+            # See Note in _get_component_ref.
+            component_description = rule_use.component
+            component_ref = _uuid_component()
+            status = Status1(state='operational')
+            component = SystemComponent(
+                uuid=component_ref,
+                type=component_type,
+                title=component_title,
+                description=component_description,
+                status=status
+            )
+            self._component_map[component_ref] = component
+        return self._component_map
+
+    def _get_component_ref(self, rule_use: RuleUse) -> str:
+        """Get component reference for specified rule use."""
+        uuid = None
+        for component_ref, component in self._component_map.items():
+            if component.type != rule_use.component_type:
+                continue
+            if component.title != rule_use.component:
+                continue
+            # Note: currently title and description are the same,
+            # therefore checking description is not necessary.
+            uuid = component_ref
+            break
+        return uuid
 
-    @staticmethod
-    def set_timestamp(value: str) -> None:
-        """Set the default timestamp value."""
-        datetime.datetime.strptime(value, '%Y-%m-%dT%H:%M:%S%z')
-        ResultsMgr.timestamp = value
+    def _derive_inventory(self) -> Dict[str, InventoryItem]:
+        """Derive inventory from RuleUse list."""
+        self._inventory_map = {}
```

Comment:
A dumb question here why are we returning this when it's a class attribute? couldn't we just run this as a procedure and access it directly?

Reply:
Fixed.